core/state_transition.go

```diff
@@ -432,7 +434,12 @@ func (st *StateTransition) TransitionDb() (*ExecutionResult, error) {
 	} else {
 		// Increment the nonce for the next transaction
 		st.state.SetNonce(msg.From, st.state.GetNonce(sender.Address())+1)
-		ret, st.gasRemaining, vmerr = st.evm.Call(sender, st.to(), msg.Data, st.gasRemaining, value)
+		if !msg.Delegate {
+			ret, st.gasRemaining, vmerr = st.evm.Call(sender, st.to(), msg.Data, st.gasRemaining, value)
+		} else {
+			var contract = vm.NewContract(sender, sender, nil, st.gasRemaining)
+			ret, st.gasRemaining, vmerr = st.evm.DelegateCall(contract, st.to(), msg.Data, st.gasRemaining)
```

Comment:
This feels like a hack, and misses the "no-store" logic.

Reply:
I'm woried about `CALLVALUE` being executed through a delegate transaction. Need to make sure that returns 0. Usually delegate call return the value of the parent call, but here there is no parent call.